internal/syncd/integrations/ecr/reconciler.go

```diff
@@ -2,27 +2,57 @@ package ecr
 
 import (
 	"context"
+	"fmt"
+	"path/filepath"
 
 	"ship-it/internal"
 
-	"github.com/google/go-github/github"
+	"github.com/google/go-github/v26/github"
+	"github.com/pkg/errors"
+	"gopkg.in/yaml.v2"
+	"ship-it/pkg/apis/k8s.wattpad.com/v1alpha1"
 )
 
 type RepositoriesService interface {
-	GetContents(ctx context.Context, org, repo, path string, opts *github.RepositoryContentGetOptions) (*github.RepositoryContent, []*github.RepositoryContent, *github.Response, error)
-	UpdateFile(ctx context.Context, owner, repo, path string, opt *github.RepositoryContentFileOptions) (*github.RepositoryContentResponse, *github.Response, error)
+	UpdateFile(msg string, branch string, path string, fileContent []byte) (*github.RepositoryContentResponse, error)
+	GetFile(branch string, path string) (string, error)
 }
 
 type ImageReconciler struct {
-	// TODO
+	Org               string
+	RegistryChartPath string
+	Branch            string
+	RepoService       RepositoriesService
 }
 
-func NewReconciler(org string, r RepositoriesService) *ImageReconciler {
+func NewReconciler(org string, prefix string, branch string, r RepositoriesService) *ImageReconciler {
 	return &ImageReconciler{
-		// TODO
+		Org:               org,
+		RegistryChartPath: prefix,
+		Branch:            branch,
+		RepoService:       r,
 	}
 }
 
 func (r *ImageReconciler) Reconcile(ctx context.Context, image *internal.Image) error {
-	return nil
+	path := filepath.Join(r.RegistryChartPath, image.Repository+".yaml")
```

Comment:
Here is where we would use the upcoming image repositories informer to build this file name. For each release affected by the updated image, we need to update each of the release `.yaml` files.